tests/base_real_sensor_mock_test_case.py

```diff
@@ -130,5 +132,16 @@ async def read_next(self, timeout: float = STD_TIMEOUT) -> str:
             A plain text or byte encoded string representing the output of the
             sensor.
         """
-        self._read_task = asyncio.Future()
-        return await asyncio.wait_for(self._read_task, timeout=timeout)
+        self._read_event.clear()
+        self.log.debug("Waiting for read event.")
+        await asyncio.wait_for(self._read_event.wait(), timeout=timeout)
+
+    async def read_until_async(self, expected: bytes) -> bytes:
```

Comment:
Please add a doc string explaining what this does and what it is for.

Given all the RpiSerialHat methods override in the unit test, I wonder if it would be cleaner to make a mock AioSerial class and add a simulation mode to RpiSerialHat that uses it. That would keep all the related code together.

Reply:
Done adding a doc string. Not a bad idea regarding the mock AioSerial class. I'll consider adding that in a future ticket.